ogre/src/OgreGpuRays.cc

```diff
@@ -196,10 +199,31 @@ void OgreGpuRays::Destroy()
     this->dataPtr->matSecondPass = nullptr;
   }
 
-  if (this->scene && this->dataPtr->orthoCam)
+  if (this->scene)
   {
-    this->scene->OgreSceneManager()->destroyCamera(this->dataPtr->orthoCam);
-    this->dataPtr->orthoCam = nullptr;
+    if (this->dataPtr->orthoCam)
+    {
+      this->scene->OgreSceneManager()->destroyCamera(this->dataPtr->orthoCam);
+      this->dataPtr->orthoCam = nullptr;
+    }
+    if (this->dataPtr->ogreCamera)
+    {
+      this->scene->OgreSceneManager()->destroyCamera(this->dataPtr->ogreCamera);
+      this->dataPtr->ogreCamera = nullptr;
+    }
+    if (this->dataPtr->visual)
+    {
+      this->scene->DestroyNode(this->dataPtr->visual);
```

Comment:
`this->dataPtr->visual == nullptr`?

Reply:
moved the `this->dataPtr->visual.reset()` call from later on in this function to inside this if check.[0413115](https://github.com/gazebosim/gz-rendering/pull/817/commits/0413115aa346c57d6d8c8dd0b8b292086497ff13)